src/data_provider/tests/sysInfoHardwareMac/sysInfoHardwareMac_test.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015, Wazuh Inc.
+ * May 18, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "sysInfoHardwareMac_test.h"
+#include "hardware/factoryHardwareFamilyCreator.h"
+#include "hardware/hardwareWrapperInterface.h"
+#include "json.hpp"
+
+void SysInfoHardwareMacTest::SetUp() {};
+
+void SysInfoHardwareMacTest::TearDown() {};
+
+using ::testing::Return;
+
+class OSHardwareWrapperMacMock: public IOSHardwareWrapper
+{
+    public:
+        OSHardwareWrapperMacMock() = default;
+        virtual ~OSHardwareWrapperMacMock() = default;
+        MOCK_METHOD(std::string, boardSerial, (), (const override));
+        MOCK_METHOD(std::string, cpuName, (), (const override));
+        MOCK_METHOD(int, cpuCores, (), (const override));
+        MOCK_METHOD(double, cpuMhz, (), (override));
+        MOCK_METHOD(uint64_t, ramTotal, (), (const override));
+        MOCK_METHOD(uint64_t, ramFree, (), (const override));
+        MOCK_METHOD(uint64_t, ramUsage, (), (const override));
+};
+
+TEST_F(SysInfoHardwareMacTest, Test_BuildHardwareData_Succeed)
```

Comment:
Do we not have some unhappy tests for this case?

Reply:
I don't realize a way to make an unhappy situation test, since the method "buildHardwareData" almost has no logic.